pkgs/development/compilers/dtc/default.nix

```diff
@@ -110,11 +110,23 @@ stdenv.mkDerivation (finalAttrs: {
     # https://github.com/NixOS/nixpkgs/pull/118700#issuecomment-885892436
     !stdenv.isDarwin &&
 
+    # tests attempt to link .so files
+    # https://github.com/dgibson/dtc/commit/dcef5f834ea34bcb9f8d0e86db1268fde52ead77
+    !stdenv.hostPlatform.isStatic &&
```

Comment:
This was fixed recently in master. You should be able to drop it (with a rebase?).